Feature/design system share page - ticket#91

vmilitaru opened the pull request:
Updates the `<ONSShare/>` component -  uses variables: `pageTitle` / `pageURL` / `title` .

Updates the component on the design system page as per the [ons design system share page example](https://ons-design-system.netlify.app/components/share-page/).

![Screenshot 2021-11-19 at 14 51 44](https://user-images.githubusercontent.com/70764326/142642937-8531131c-92d3-46b0-924a-35e13bf1f607.png)

Adds and updates the share component on the components page to match the census atlas design.
 
![Screenshot 2021-11-19 at 14 51 28](https://user-images.githubusercontent.com/70764326/142643123-9446c04b-97ce-4349-9e50-1262ed270e1b.png)

designorant reviewed (changes requested):
Just for the record as we discussed this on Slack, let's make the component as agnostic as possible while trying to stay as close to the original as possible:
- remove the invalid HTML wrapper specific to landing page
- introduce a list modifier that will fake 2-column layout via CSS overrides on the list item (this should ideally be one off per page thing, but we're using ONSShare in quite a few places hence the modifier route)

I think we should have a global Icon component but let's look into that later.

vmilitaru commented:
> Just for the record as we discussed this on Slack, let's make the component as agnostic as possible while trying to stay as close to the original as possible:
> 
> * remove the invalid HTML wrapper specific to landing page
> * introduce a list modifier that will fake 2-column layout via CSS overrides on the list item (this should ideally be one off per page thing, but we're using ONSShare in quite a few places hence the modifier route)
> 
> I think we should have a global Icon component but let's look into that later.

I've updated the share page component, and I've added the modifier on the list item, not on the list .... but please have a look and let me know if further changes are needed .

designorant reviewed (approved):
Cheers!
